packages/experience/src/hooks/use-single-sign-on-watch.ts

```diff
@@ -1,31 +1,30 @@
 import { SignInIdentifier, type SsoConnectorMetadata } from '@logto/schemas';
-import { useEffect, useState, useCallback, useContext } from 'react';
-import { type Control, useWatch } from 'react-hook-form';
+import { useEffect, useCallback, useContext } from 'react';
 import { useNavigate } from 'react-router-dom';
 
 import SingleSignOnContext from '@/Providers/SingleSignOnContextProvider/SingleSignOnContext';
+import SingleSignOnFormModeContext from '@/Providers/SingleSignOnFormModeContextProvider/SingleSignOnFormModeContext';
 import { getSingleSignOnConnectors } from '@/apis/single-sign-on';
+import type { IdentifierInputValue } from '@/components/InputFields/SmartInputField';
 import { singleSignOnPath } from '@/constants/env';
 import useApi from '@/hooks/use-api';
 import useSingleSignOn from '@/hooks/use-single-sign-on';
 import { validateEmail } from '@/utils/form';
 
-import type { FormState } from './index';
+import { useSieMethods } from './use-sie';
 
-const useSingleSignOnWatch = (control: Control<FormState>) => {
+const useSingleSignOnWatch = (identifierInput?: IdentifierInputValue) => {
```

Comment:
move the formState watcher to the upper scope. Different form may have different form status type. This hook only cares about the identifier input value.